src/main/java/org/jooq/lambda/Seq.java

```diff
@@ -3397,6 +3397,30 @@ public R next() {
     }
 
     /**
+     * Chunk a Stream based on a predicate
+     * <p>
+     * <code><pre>
+     * // ((1,1,2), (2), (3,4))
+     * Seq.of(1, 1, 2, 2, 3, 4).chunked(n -> n %2 == 0)
```

Comment:
> Should we assume the beginning chunk starts true? or should this actually return ((2), (2, 3), (4)) skipping anything before the first chunk.

Egh... :) OK, so in the no chunk case, we get:

- 1 chunk for 0 true evaluations of the predicate
- 2 chunks for 1 true evaluation of the predicate
- 3 chunks for 2 true evaluations of the predicate

For example:

```java
// ((1, 1, 2, 2, 3, 4))
Seq.of(1, 1, 2, 2, 3, 4).chunked(i -> false);

// ((),(),(),(),(),(),())
Seq.of(1, 1, 2, 2, 3, 4).chunked(i -> true);
```

I think that options 1-3 should adhere to this logic. My example 1 was wrong, there should be a trailing empty chunk. Will fix the comment.